### PR TITLE
Locally print to TTY

### PR DESCRIPTION
**Pull request by illusionyy**

### CODE REVIEW

**Comment by bucanero** [comment]
No need for the `#ifdef __PS4__`, it can be done the right way:

if you want to add a TTY output, then you should define a new method , TTY_LOGGER, and then handle that case as needed.

then this new output method will be available for all platforms (PS3, vita, PS4)

File: source/dbglogger.c [hunk 1]
```diff
@@ -258,7 +258,10 @@ void dbglogger_log(const char* fmt, ...) {
         va_start(arg, fmt);
         vsnprintf(buffer, sizeof(buffer), fmt, arg);
         va_end(arg);
-    
+
+#ifdef __PS4__
+        puts(buffer);
+#endif
```
Comment (bucanero) [inline source/dbglogger.c]: let's avoid this and add a `TTY_LOGGER` output if needed.

**Review by bucanero — CHANGES_REQUESTED** [review]

File: source/dbglogger.c (hunk 1 above)
Comment (illusionyy) [reply source/dbglogger.c]: Added `__USE_TTY_LOG__`

File: include/dbglogger.h [hunk 2]
```diff
@@ -37,6 +37,11 @@ extern "C" {
 //  TCP example string    "tcp:192.168.1.123:18194"
 //  File example string   "file:/dev_hdd0/temp/app.log"
 
+#ifdef (__USE_TTY_LOG__)
+#define TTY_LOGGER puts
+#else
+#define TTY_LOGGER(...)
+#endif
```
Comment (bucanero) [inline include/dbglogger.h]: No, I mean you should add a `TTY_LOGGER` option to the LOGGER_MODES `typedef enum`

then handle the case inside the switch(loggerMode) of `dbglogger_printf`

File: source/dbglogger.c [hunk 3]
```diff
@@ -237,6 +237,7 @@ void dbglogger_printf(const char* fmt, ...) {
         vsnprintf(buffer, sizeof(buffer), fmt, arg);
         va_end(arg);
     
+        TTY_LOGGER(buffer);
```
Comment (bucanero) [inline source/dbglogger.c]: handle the case inside the switch(loggerMode) of `dbglogger_printf`

**Review by bucanero — CHANGES_REQUESTED** [review]

**Comment by bucanero** [comment]
@illusion0001 did you had a chance to look into this?

if not, no worries, just open an issue to "add support for TTY logging" and I'll take a look when I have a chance. Then I'll add this feature to the master branch.

**Comment by illusionyy** [comment]
Haven't had a chance to look into this, misunderstood the previous comment 😅.

**Comment by illusionyy** [comment]
Hopefully I got it this time. Added `TTY_LOGGER` in switch cases

File: source/dbglogger.c [hunk 4]
```diff
@@ -316,6 +324,9 @@ int dbglogger_init_str(const char* ini_str) {
     } else 
     if (strcmp(mode, FILE_INI_STR) == 0) {
         return dbglogger_init_mode(FILE_LOGGER, data, 0);
+    } else
+    if (strcmp(mode, TTY_INI_STR) == 0) {
+        return dbglogger_init_mode(TTY_INI_STR, data, 0);
```
Comment (bucanero) [inline source/dbglogger.c]:
```suggestion
        return dbglogger_init_mode(TTY_LOGGER, NULL, 0);
```

**Comment by bucanero** [comment]
> Hopefully I got it this time. Added `TTY_LOGGER` in switch cases

cool 👍 
now you should be able to use the TTY logger from Cheats Manager or Apollo, by using:

```c
dbglogger_init_mode(TTY_LOGGER, NULL, 0);
```

just place that line at the begging of main() in `main.c`, and you'll get debug logs directly on TTY.

**Review by bucanero — APPROVED** [review]

**Pull request merged**